File: java/04-pgm-contador/pgmContador.java
import java.util.Date;
import java.lang.Exception;
import java.lang.Thread;

public class pgmContador{
    public static void main(String[] args) {
        System.out.println("+--------------------------------------------+");
        System.out.println("+------------- PGM: PgmContador -------------+");
        System.out.println("+Autor:👉🏽👉🏽👉🏽Santiago Duran👈🏽👈🏽👈🏽+");
        System.out.println("+--------------------------------------------+");
        
        try {
            System.out.println("Contando desde el shell a las "+new Date());
            for (int i = 0; i < 20; i++) {
                System.out.println("Posicion Indice ❇️: "+i);
                Thread.sleep(1000-(i*50));
            }
            System.out.println("Contador Finalizado...🔥🔥🔥🔥🔥");
        } catch (Exception e) {
            System.err.println("⚠️⚠️⚠️ "+e);
        }
        
    }
}

File: c++/04-pgm-contador/pgmContador.cpp
#include <iostream>
#include <ctime>
#include <chrono>
#include <thread>

using namespace std;

int main() {
    cout << "+--------------------------------------------+" << endl;
    cout << "+------------- PGM: PgmContador -------------+" << endl;
    cout << "+Autor:👉🏽👉🏽👉🏽Santiago Duran👈🏽👈🏽👈🏽+" << endl;
    cout << "+--------------------------------------------+" << endl;

    try {
        auto now = chrono::system_clock::now();
        time_t current_time = chrono::system_clock::to_time_t(now);
        cout << "Contando desde el shell a las " << ctime(&current_time);

        for (int i = 0; i < 20; i++) {
            cout << "Posicion Indice ❇️: " << i << endl;
            this_thread::sleep_for(chrono::milliseconds(1000-(i*50)));
        }
        cout << "Contador Finalizado...🔥🔥🔥🔥🔥" << endl;
    } catch (exception& e) {
        cerr << "⚠️⚠️⚠️ " << e.what() << endl;
    }
    
    return 0;
}
